modules/videopress/utility-functions.php

```diff
@@ -678,12 +678,15 @@ function videopress_get_attachment_url( $post_id ) {
 	if ( ! isset( $meta['videopress']['files']['hd']['mp4'] ) ) {
 		// Use the original file as the url if it isn't transcoded yet.
 		if ( isset( $meta['original'] ) ) {
-			return $meta['original'];
+			$return = $meta['original'];
+		} else {
+			// Otherwise, there isn't much we can do.
+			return null;
 		}
-
-		// Otherwise, there isn't much we can do.
-		return null;
+	} else {
+		$return = $meta['videopress']['file_url_base']['https'] . $meta['videopress']['files']['hd']['mp4'];
 	}
 
-	return $meta['videopress']['file_url_base']['https'] . $meta['videopress']['files']['hd']['mp4'];
+	// If the URL is a string, return it. Otherwise, we shouldn't to avoid errors downstream, so null.
+	return ( is_string( $return ) ) ? $return : null;
```

Comment:
Should we consider `esc_url` the result here, since the result should always be a URL?

Reply:
`esc_url()` also html-encodes. If we want to do that, we should probably use `esc_url_raw()`, which does not html-encode. (Unless the data is already html-encoded, in which case it may get a bit more complicated.)

Reply:
But should we even encode here? We don't know what context that string is going to be used in. Plus it has been unencoded previously, I'm betting it's used somewhere with `esc_url`.
Aaand I decided to get off my couch and look. It's used as a filter for `wp_get_attachment_url`: 
https://github.com/Automattic/jetpack/blob/master/modules/videopress/class.jetpack-videopress.php#L43
Which in turn looks to be used with `esc_url` in the end.

Reply:
```Sanitize early.
Escape late.
Drink [coffee] often.```

Yeah, we should be fine without encoding, but there's been some attempts to pass an object along through here. I couldn't duplicate how to create that situation, but ensuring we only pass strings would prevent fatals.